src/daft-scan/src/scan_task_iters.rs

```diff
@@ -111,3 +116,109 @@ impl Iterator for MergeByFileSize {
+                    (
+                        FileFormatConfig::Parquet(_),
+                        StorageConfig::Native(_),
+                        [source],
+                        Some(None),
+                        None,
+                    ) if source
+                        .get_size_bytes()
+                        .map_or(true, |s| s > split_threshold_bytes as u64) =>
+                    {
+                        let (io_runtime, io_client) =
+                            t.storage_config.get_io_client_and_runtime()?;
+
+                        let path = source.get_path();
+
+                        let io_stats =
+                            IOStatsContext::new(format!("split_by_row_groups for {:#?}", path));
+
+                        let runtime_handle = io_runtime.handle();
+
+                        let file = runtime_handle.block_on(read_parquet_metadata(
+                            path,
+                            io_client,
+                            Some(io_stats),
+                        ))?;
+
+                        let mut new_tasks: Vec<DaftResult<ScanTaskRef>> = Vec::new();
+                        let mut curr_row_groups = Vec::new();
+                        let mut curr_size_bytes = 0;
+
+                        for (i, rg) in file.row_groups.iter().enumerate() {
+                            curr_row_groups.push(i as i64);
+                            curr_size_bytes += rg.compressed_size();
+
+                            if curr_size_bytes >= min_size_bytes || i == file.row_groups.len() - 1 {
+                                let mut new_source = source.clone();
+                                match &mut new_source {
+                                    DataFileSource::AnonymousDataFile {
+                                        chunk_spec,
+                                        size_bytes,
+                                        ..
+                                    }
+                                    | DataFileSource::CatalogDataFile {
+                                        chunk_spec,
+                                        size_bytes,
+                                        ..
+                                    } => {
+                                        *chunk_spec = Some(ChunkSpec::Parquet(curr_row_groups));
+                                        *size_bytes = Some(curr_size_bytes as u64);
+                                    }
+                                };
+
+                                new_tasks.push(Ok(ScanTask::new(
+                                    vec![new_source],
+                                    t.file_format_config.clone(),
+                                    t.schema.clone(),
+                                    t.storage_config.clone(),
+                                    t.pushdowns.clone(),
+                                )
+                                .into()));
+
+                                curr_row_groups = Vec::new();
```

Comment:
Nit: clearing the existing vec will probably be more efficient, since it can reuse the existing allocation.
```suggestion
                                curr_row_groups.clear()
```

Reply:
Ah ok. That makes sense. `curr_row_groups` does get moved when initializing the chunk spec, so I would need to clone it for that. Would the tradeoff still be beneficial?

Reply:
Ah if there's a move happening above, then reinitializing here is fine! Cloning + clearing should be ~the same in terms of performance and memory utilization, but is a bit less readable than a move + reinitialize IMO, so I think that we can leave this as-is.